app/controllers/api/messages_controller.rb

```diff
@@ -1,33 +1,93 @@
 class Api::MessagesController < Api::TenantController
   before_action :set_en_locale
-  before_action :check_content_type, only: :create
+  before_action :load_box, only: :message_drafts
+  before_action :check_message_type, only: :message_drafts
 
-  include Upvs::MessageDraftConcern
-
-  ALLOWED_CONTENT_TYPES = ['application/json;type=upvs']
-
-  def create
-    case message_type(raw_content_type)
-    when 'upvs'
-      create_upvs_message_draft
-    end
-  end
+  ALLOWED_MESSAGE_TYPES = ['Upvs::MessageDraft']
 
   def show
     @message = @tenant.messages.find(params[:id])
   end
 
+  def message_drafts
+    render_unprocessable_entity('Invalid sender') and return unless @box
+
+    ::Message.transaction do
+      @message = permitted_params[:type].classify.safe_constantize.load_and_validate(permitted_params, box: @box)
```

Comment:
metody ktore robia viac veci naraz su trosku smell, nestaci zavolat to valid? nizsie? ci to su ine validacie?

Reply:
Upravene.